src/Exercism.Representers.CSharp/Normalization/NormalizeDictionaryInitialization.cs

```diff
@@ -0,0 +1,183 @@
+        {
+            if (semanticModel == default)
+            {
+                Log.Error(
+                    $"{nameof(NormalizeDictionaryInitialization)}: semantic model is null - this normalization will fail");
+            }
+
+            this.semanticModel = semanticModel;
+        }
+
+        public override SyntaxNode VisitInitializerExpression(InitializerExpressionSyntax node)
+        {
+            SyntaxNode DefaultVisit() => base.VisitInitializerExpression(node);
+
+            if (semanticModel == default)
+            {
+                return DefaultVisit();
+            }
+
+            try
+            {
+                if (semanticModel.GetOperation(node) is IObjectOrCollectionInitializerOperation initializerOperation)
+                {
+                    if (initializerOperation.Parent?.Type?.Name != "Dictionary")
+                    {
+                        if (initializerOperation.Parent?.Type?.Name == default)
+                        {
+                            Log.Error(
+                                $"{nameof(NormalizeDictionaryInitialization)}: unable to retrieve the type information for {nameof(initializerOperation)}");
+                        }
+
+                        return DefaultVisit();
+                        // presumably a list or some object or other
+                    }
+
+                    var initializerExtractionResults
+                        = node.Kind() switch
+                        {
+                            SyntaxKind.CollectionInitializerExpression
+                            => ExtractInitializersWithCollectionSyntax(initializerOperation),
+                            SyntaxKind.ObjectInitializerExpression
+                            => ExtractInitializersWithObjectSyntax(initializerOperation),
+                            _ 
+                            => throw new InvalidKindException{Kind = node.Kind()}
+                        };
+
+                    if (!initializerExtractionResults.Success)
+                    {
+                        return DefaultVisit();
+                    }
+
+                    var visitedInitializerSyntaxNodes = initializerExtractionResults.InitializerSyntaxNodes
+                        .Select(initializer => new KeyValuePair<SyntaxNode, SyntaxNode>(
+                            this.Visit(initializer.Key),
+                            this.Visit(initializer.Value))
+                        );
+
+                    var dictionaryAsText = BuildDictionaryAsText(visitedInitializerSyntaxNodes);
+                    var dictionarySyntaxTree = CSharpSyntaxTree.ParseText(dictionaryAsText);
+                    var replacementNode = dictionarySyntaxTree
+                        .GetRoot()
+                        .DescendantNodes()
+                        .FirstOrDefault(n => n is InitializerExpressionSyntax);
+                    if (replacementNode == default)
```

Comment:
When will this be true?

Reply:
Never.  It's just part of that billion dollar mistake.

Reply:
@ErikSchierboom We could have a shot at building with nullable enabled which would satisfy my need not to lay traps for maintainers.  It looks very doable.   What do think?

Reply:
Go for it! I love static analysis.